pkg/arvo/app/hark-store.hoon

```diff
@@ -596,6 +598,16 @@
     =>  (emit autoseen-timer)
     poke-core(current-timebox now.bowl)
   ::
+  ++  read-all
+    =.  unreads-count
+      %-  ~(run by unreads-count)
+      |*(* 0)
+    =.  poke-core  seen
+    %-  ~(rep ^in ~(key by unreads-each))
+    |=  [st=stats-index:store pok=$_(poke-core)]
```

Comment:
This should also be a `run:by`

Reply:
i need to accumulate the updated `poke-core` though, for each key in unreads-each. `run:by` would give me a list of `poke-cores`,  no?

Reply:
sorry for the late reply!

```hoon
 ++  read-all
    =:  unreads-count  (~(run by unreads-count) _0)
        unreads-each    (~(run by unreads-each) _~)
      ==
   seen
```

Reply:
oof. much better. thanks!